bokehjs/src/lib/models/expressions/stack.ts

```diff
@@ -29,6 +29,9 @@ export class Stack extends Expression {
   protected _v_compute(source: ColumnarDataSource): Arrayable<number> {
     const result = new Float64Array(source.get_length() || 0)
     for (const f of this.fields) {
+      if (Object.keys(source.data).length == 0) {
+        break;
+      }
```

Comment:
This should be moved out one level. However, a better approach would be to check if a field `f` exists on `source.data`. This way partially initialized data sources would still be usable (if that makes sense), or at least the wouldn't fail.

```ts
const n = source.get_length() || 0
const result = new Float64Array(n)
for (const f of this.fiels) {
  const column = source.data[f]
  if (column != null) {
    for (let i = 0, k = Math.min(n, column.length); i < k; i++) {
     result[i] += column[i]
    }
  }
  return result
}
```

Reply:
@kennedydane will you be able to make the change suggested above?